src/taco-utils/test/utilHelper.ts

```diff
@@ -163,4 +173,70 @@ describe("UtilHelper", function (): void {
             parsed.remain[1].should.equal("foo");
         });
     });
+
+    describe("parseUserJSON()", function (): void {
+        var testHome: string = path.join(os.tmpdir(), "taco-utils", "parseUserJSON");
+        before(function (): void {
+            process.env["TACO_HOME"] = testHome;
+            rimraf.sync(testHome);
+            mkdirp.sync(testHome);
+        });
+
+        after(function (): void {
+            rimraf(testHome, function (err: Error): void {/* ignored */ }); // Not sync, and ignore errors
+        });
+
+        it("read UTF-16 file", function (): void {
+            writeBufferAndValidate("utf16.json", (stringified: string) => new Buffer(stringified, "utf16le"));
+        });
+
+        it("read UTF-8 file", function (): void {
+            writeBufferAndValidate("utf8.json", stringified => new Buffer(stringified, "utf8"));
+        });
+
+        it("read UTF-8 with BOM", function (): void {
+            var parsedJson = UtilHelper.parseUserJSON(path.resolve(__dirname, "taco_utf8BOM.json"));
+            parsedJson["cordova-cli"].should.equal("5.3.3");
+        });
+
+        it("read file which doesn't exist", function(): void {
+            try {
+                UtilHelper.parseUserJSON(path.resolve(__dirname, "nonExistentFile.json"));
+            } catch (e) {
+                (<TacoError>e).errorCode.should.equal(TacoErrorCodes.ErrorUserJsonMissingOrMalformed);
+            }
+        });
+
+        it("read malformed file", function (): void {
+            writeBufferAndValidate("utf8.json", stringified => {
+                var buffer: Buffer = new Buffer(stringified, "utf8");
+                var malformedBuffer: Buffer = new Buffer(buffer.length + 3);
+                // add some random bytes to the malformed buffer
+                malformedBuffer.write("\u00bd\u00bc\u00be");
+                buffer.copy(malformedBuffer, malformedBuffer.length, 0);
+                return malformedBuffer;
+            }, TacoErrorCodes.ErrorUserJsonMissingOrMalformed);
+        });
+
+        function writeBufferAndValidate(filename: string, getBuffer: (stringified: string) => Buffer, expectedErrorCode?: TacoErrorCodes) {
+            var x = { "cordova-cli": 5.8 };
+            var stringified: string = JSON.stringify(x);
+            var buffer: Buffer = getBuffer(stringified);
+
+            var filepath = path.join(testHome, filename);
+            fs.writeFileSync(filepath, buffer);
+            try {
+                var parsedJson = UtilHelper.parseUserJSON(filepath);
+                if (!expectedErrorCode) {
+                    JSON.stringify(parsedJson).should.equal(stringified);
```

Comment:
I think that should.equal also works properly for JSON objects, so you might be able to do parsedJson.should.equal(x)